tests/components/sleepiq/test_init.py

```diff
@@ -1,65 +1,54 @@
-    base_url = "https://prod-api.sleepiq.sleepnumber.com/rest/"
-    if single:
-        suffix = "-single"
-    else:
-        suffix = ""
-    mock.put(base_url + "login", text=load_fixture("sleepiq-login.json"))
-    mock.get(base_url + "bed?_k=0987", text=load_fixture(f"sleepiq-bed{suffix}.json"))
-    mock.get(base_url + "sleeper?_k=0987", text=load_fixture("sleepiq-sleeper.json"))
-    mock.get(
-        base_url + "bed/familyStatus?_k=0987",
-        text=load_fixture(f"sleepiq-familystatus{suffix}.json"),
-    )
-
-
-async def test_setup(hass, requests_mock):
-    """Test the setup."""
-    mock_responses(requests_mock)
-
-    # We're mocking the load_platform discoveries or else the platforms
-    # will be setup during tear down when blocking till done, but the mocks
-    # are no longer active.
-    with patch("homeassistant.helpers.discovery.load_platform", MagicMock()):
-        assert sleepiq.setup(hass, CONFIG)
-
-
-async def test_setup_login_failed(hass, requests_mock):
-    """Test the setup if a bad username or password is given."""
-    mock_responses(requests_mock)
-    requests_mock.put(
-        "https://prod-api.sleepiq.sleepnumber.com/rest/login",
-        status_code=HTTPStatus.UNAUTHORIZED,
-        json=load_fixture("sleepiq-login-failed.json"),
-    )
-
-    response = sleepiq.setup(hass, CONFIG)
-    assert not response
-
-
-async def test_setup_component_no_login(hass):
-    """Test the setup when no login is configured."""
-    conf = CONFIG.copy()
-    del conf["sleepiq"]["username"]
-    assert not await setup.async_setup_component(hass, sleepiq.DOMAIN, conf)
-
-
-async def test_setup_component_no_password(hass):
-    """Test the setup when no password is configured."""
-    conf = CONFIG.copy()
-    del conf["sleepiq"]["password"]
-
-    assert not await setup.async_setup_component(hass, sleepiq.DOMAIN, conf)
+"""Tests for the SleepIQ integration."""
+from unittest.mock import patch
+
+from homeassistant.components.sleepiq.const import DOMAIN
+from homeassistant.components.sleepiq.coordinator import UPDATE_INTERVAL
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.util.dt import utcnow
+
+from tests.common import async_fire_time_changed
+from tests.components.sleepiq.conftest import (
+    mock_bed_family_status,
+    mock_beds,
+    mock_sleepers,
+)
+
+
+async def test_unload_entry(hass: HomeAssistant, setup_entry) -> None:
+    """Test unloading the SleepIQ entry."""
+    entry = setup_entry["mock_entry"]
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.state is ConfigEntryState.NOT_LOADED
+    assert not hass.data.get(DOMAIN)
+
+
+async def test_entry_setup_login_error(hass: HomeAssistant, config_entry) -> None:
+    """Test when sleepyq client is unable to login."""
+    with patch("sleepyq.Sleepyq.login", return_value=False):
```

Comment:
```suggestion
    with patch("sleepyq.Sleepyq.login", side_effect=ValueError):
```

That will get you to 100%. `Sleepyq.login` actually never returns False.